### PR TITLE
Updated getting and setting data in state for deep nested data

### PR DESCRIPTION
**Pull request by emeidell**

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -52,24 +54,20 @@ module.exports = function () {
   }, {
     key: "getFieldFormValue",
     value: function getFieldFormValue(attrName) {
-      if (isArray(this.parsedData[attrName])) {
-        return this.parsedData[attrName];
-      } else if (isEmpty(this.formData[attrName]) && !isNumber(this.formData[attrName])) {
+      var data = get(this.formData, attrName);
```
Comment (emeidell) [inline dist/index.js]: This "get" and "set" are the backbones to deep nested data in the FormLinker. Gives the ability to set and fetch data nested deep without having to iterate over the data sets a whole bunch.

File: index.js [hunk 2]
```diff
@@ -35,23 +37,19 @@ module.exports = class{
   }
 
   getFieldFormValue(attrName) {
-    if(isArray(this.parsedData[attrName])) {
-      return(this.parsedData[attrName]);
-    } else if(isEmpty(this.formData[attrName]) && !isNumber(this.formData[attrName])) {
+    let data = get(this.formData, attrName);
```
Comment (emeidell) [inline index.js]: This "get" and "set" are the backbones to deep nested data in the FormLinker. Gives the ability to set and fetch data nested deep without having to iterate over the data sets a whole bunch.

File: dist/index.js [hunk 3]
```diff
@@ -7,12 +7,14 @@ function _defineProperty(obj, key, value) { if (key in obj) { Object.definePrope
...
 
 var isArray = require("lodash/isArray");
+var isBoolean = require("lodash/isBoolean");
 var isEqual = require("lodash/isEqual");
 var isEmpty = require("lodash/isEmpty");
 var isNil = require("lodash/isNil");
 var isNumber = require("lodash/isNumber");
 var isObject = require("lodash/isObject");
 var set = require("lodash/set");
+var get = require("lodash/get");
```
Comment (travispcook) [inline dist/index.js]: i forget that its not react, so i was confused at first why you set these to a var. but i get it now.

File: dist/index.js (hunk 1 above)
Comment (travispcook) [inline dist/index.js]: so simple. thats cool.

File: test/linker.js [hunk 4]
```diff
@@ -11,3 +11,107 @@ test("test", t => {
...
+});
+
+test("Is a Boolean", t => {
+  let fl = new FormLinker({
+    data: {
+      foo: {
+        bar: 42
+      }
+    }
+  });
```
Comment (travispcook) [inline test/linker.js]: yay for test cases!

**Pull request closed without merging**